src/components/structures/RoomSearch.tsx

```diff
@@ -43,220 +34,49 @@ interface IProps {
-        if (this.state.spotlightBetaEnabled !== spotlightBetaEnabled) {
-            this.setState({
-                spotlightBetaEnabled,
-                query: "",
-            });
-        }
-        // in case the user was in settings at the 5-minute mark, dismiss the toast
-        ToastStore.sharedInstance().dismissToast("BETA_SPOTLIGHT_TOAST");
-    };
-
     private openSpotlight() {
         Modal.createDialog(SpotlightDialog, {}, "mx_SpotlightDialog_wrapper", false, true);
     }
 
     private onAction = (payload: ActionPayload) => {
-        if (payload.action === Action.ViewRoom && payload.clear_search) {
-            this.clearInput();
-        } else if (payload.action === 'focus_room_filter') {
-            if (this.state.spotlightBetaEnabled) {
-                this.openSpotlight();
-            } else {
-                this.focus();
-            }
-        }
-    };
-
-    private clearInput = () => {
-        if (this.elementRef.current?.tagName !== "INPUT") return;
-        (this.elementRef.current as HTMLInputElement).value = "";
-        this.onChange();
-    };
-
-    private openSearch = () => {
-        if (this.state.spotlightBetaEnabled) {
+        if (payload.action === 'focus_room_filter') {
             this.openSpotlight();
-        } else {
-            // dispatched as it needs handling by MatrixChat too
-            defaultDispatcher.dispatch({ action: "focus_room_filter" });
         }
     };
 
-    private onChange = () => {
-        if (this.elementRef.current?.tagName !== "INPUT") return;
-        this.setState({ query: (this.elementRef.current as HTMLInputElement).value });
-    };
-
-    private onFocus = (ev: React.FocusEvent<HTMLInputElement>) => {
-        this.setState({ focused: true });
-        ev.target.select();
-    };
-
-    private onBlur = (ev: React.FocusEvent<HTMLInputElement>) => {
-        this.setState({ focused: false });
-    };
-
-    private onKeyDown = (ev: React.KeyboardEvent) => {
-        const action = getKeyBindingsManager().getRoomListAction(ev);
-        switch (action) {
-            case KeyBindingAction.ClearRoomFilter:
-                this.clearInput();
-                defaultDispatcher.fire(Action.FocusSendMessageComposer);
-                break;
-            case KeyBindingAction.SelectRoomInRoomList: {
-                const shouldClear = this.props.onSelectRoom();
-                if (shouldClear) {
-                    // wrap in set immediate to delay it so that we don't clear the filter & then change room
-                    setImmediate(() => {
-                        this.clearInput();
-                    });
-                }
-                break;
-            }
-        }
-    };
-
-    public focus = (): void => {
-        this.elementRef.current?.focus();
-    };
-
     public render(): React.ReactNode {
         const classes = classNames({
             'mx_RoomSearch': true,
-            'mx_RoomSearch_hasQuery': this.state.query,
```

Comment:
Some of these classes doesn't seem to be used anymore but they are still around in the Sass files. Can you chat that please and eventually remove them?